pkg/review/commit_review_status_test.go

```diff
@@ -0,0 +1,454 @@
+                     "endCursor": "FG",
+                     "hasNextPage": false,
+                     "hasPreviousPage": true,
+                     "startCursor": "XQ"
+                   },
+                   "totalCount": 2
+                 }
+               }
+             }
+           }
+         }`,
+			},
+		},
+		{
+			name:       "no_associated_pull_requests_for_a_commit",
+			token:      "ghp_JQqAJky0GlNB7xqVCbUrivIgFZ09V8gGCi5C",
+			githubOrg:  "test-org",
+			repository: "test-repo",
+			commitSha:  "kof6p96lr6qvdu81qw49fhmoxrod9qmc2qak51nh",
+			wantRequestBodies: []string{
+				`{
+           "query": "
+             query($commitSha:GitObjectID! $githubOrg:String! $pageCursor:String! $repository:String!) {
+               repository(owner: $githubOrg, name: $repository) {
+                 object(oid: $commitSha) {
+                   ... on Commit{
+                     associatedPullRequests(first: 100, after: $pageCursor) {
+                       nodes{
+                         databaseId,
+                         number,
+                         reviewDecision
+                       },
+                       pageInfo{
+                         endCursor,
+                         hasNextPage,
+                         hasPreviousPage,
+                         startCursor
+                       },
+                       totalCount
+                     }
+                   }
+                 }
+               }
+             }
+           ",
+           "variables": {
+             "commitSha": "kof6p96lr6qvdu81qw49fhmoxrod9qmc2qak51nh",
+             "githubOrg": "test-org",
+             "pageCursor": "",
+             "repository":"test-repo"
+           }
+         }`,
+			},
+			want: []*PullRequest{},
+			responseBodies: []string{
+				`{
+           "data": {
+             "repository": {
+               "object": {
+                 "associatedPullRequests": {
+                   "nodes": [],
+                   "pageInfo": {
+                     "endCursor": "XQ",
+                     "hasNextPage": false,
+                     "hasPreviousPage": false,
+                     "startCursor": ""
+                   },
+                   "totalCount": 0
+                 }
+               }
+             }
+           }
+         }`,
+			},
+		},
+	}
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			gotRequestBodies := make([]string, len(tc.wantRequestBodies))
+			requestNumber := 0
+			fakeGitHub := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				expectedAuthHeader := fmt.Sprintf("Bearer %s", tc.token)
+				if r.Header.Get("Authorization") != expectedAuthHeader {
+					w.WriteHeader(500)
+					fmt.Fprintf(w, "missing or malformed authorization header")
+					return
+				}
+				bytes, _ := io.ReadAll(r.Body)
+				requestBody := string(bytes)
+				gotRequestBodies[requestNumber] = requestBody
```

Comment:
This assumes the requests you got won't exceed the size of your "want", which might not be true if you have a bug. I'd suggest simply appending to `gotRequestBodies` without an initial size.

Reply:
done.